vaadin-grid-flow-integration-tests/frontend/benchmark.js

```diff
@@ -0,0 +1,186 @@
+      to {
+        opacity: 1;
+      }
+    }
+
+    :host {
+      animation: content-ready 1s;
+    }
+  `
+);
+
+const whenRendered = (grid) => {
+  return new Promise((resolve) => {
+    let readyTimer;
+    let endTime;
+    const listener = (e) => {
+      if (e.animationName === 'content-ready' || e.propertyName === 'opacity') {
+        endTime = performance.now();
+        readyTimer && clearTimeout(readyTimer);
+        if (!grid.loading) {
+          readyTimer = setTimeout(() => {
+            grid.$.scroller.removeEventListener('animationstart', listener);
+            grid.removeEventListener('animationstart', listener);
+            grid.removeEventListener('transitionstart', listener);
+            resolve(performance.now() - endTime);
+            // The timeout needs to be large enough so everything gets rendered
+            // but small enough so the tests won't take forever. This resolves with
+            // the idle time between the last invocation and timeout.
+          }, 1000);
+        }
+      }
+    };
+
+    grid.$.scroller.addEventListener('animationstart', listener);
+    grid.addEventListener('animationstart', listener);
+    grid.addEventListener('transitionstart', listener);
+  });
+};
+
+const reportResult = (result) => {
+  // @ts-ignore
+  window.tachometerResult = result;
+
+  const resultDiv = document.createElement('div');
+  resultDiv.textContent = `Result: ${result}`;
+  resultDiv.style.position = 'fixed';
+  resultDiv.style.left = resultDiv.style.top = '0';
+  document.body.appendChild(resultDiv);
+};
+
+let start = 0;
+
+// @ts-ignore
+window.startWhenReady = () => {
+  return Promise.all([
+    customElements.whenDefined('vaadin-grid'),
+    customElements.whenDefined('vaadin-grid-column'),
+  ]).then(() => (start = performance.now()));
+};
+
+// @ts-ignore
+window.startWhenRendered = (grid) => {
+  return whenRendered(grid).then(() => {
+    start = performance.now();
+  });
+};
+
+// @ts-ignore
+window.whenRendered = whenRendered;
+
+// @ts-ignore
+window.measureRender = (grid) => {
+  whenRendered(grid).then((idleTime) => {
+    reportResult(performance.now() - start - idleTime);
+  });
+};
+
+const SCROLL_TIME = 10000;
+const WARMUP_TIME = 1000;
+
+const scroll = (grid, frames, startTime, previousTime, deltaXMultiplier, deltaYMultiplier) => {
+  const now = performance.now();
+  const e = new CustomEvent('wheel', { bubbles: true, cancelable: true });
+
+  // @ts-ignore
+  e.deltaX = (now - previousTime) * deltaXMultiplier;
+  // @ts-ignore
+  e.deltaY = (now - previousTime) * deltaYMultiplier;
+
+  grid.dispatchEvent(e);
+
+  // Switch horizontal scroll direction in case end was reached
+  if (deltaXMultiplier) {
+    const overflow = grid.getAttribute('overflow');
+    if (deltaXMultiplier === 1 && !overflow.includes('right')) {
+      deltaXMultiplier = -1;
+    } else if (deltaXMultiplier === -1 && !overflow.includes('left')) {
+      deltaXMultiplier = 1;
+    }
+  }
```

Comment:
This switching of directions didn't work for `multicolumn` variant when I tested the page locally. After reaching the right end, it stays there until the test is finished. With `mixed` variant it worked.

Reply:
That's strange. For me, http://localhost:9998/benchmark?metric=horizontalscrollframetime&variant=multicolumn worked fine on both Chrome and FF

Reply:
Hmm... I have tested with Windows Chrome and Linux Mint Chrome and FF, which all reproduce the issue.

Once the scrollbar is within ~15px from the end, events with any positive `deltaX` are dismissed.

@tulioag does this work in your environment?

Reply:
Was able to reproduce this by enabling the physical scrollbars. Fixed by having 100px thresholds on both ends instead.

Reply:
👍 seems to work